### PR TITLE
Example, test and fix to use keras

### PR DESCRIPTION
**Pull request by javierluraschi**
Add example, test and a fix to use `keras` with `cloudml`.

### CODE REVIEW

File: inst/cloudml/setup.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
     # ml-engine doesn't provide TensorFlow 1.3 yet but they could be potentially
     # upgraded; however, we've found out some components (e.g. tfestimators) hang even
     # under python when upgrading TensorFlow versions.
-    # ["pip", "install", "keras", "--upgrade"],
+    ["pip", "install", "keras", "--upgrade"]
```
Comment (jjallaire) [inline inst/cloudml/setup.py]: We need some additional dependencies to get all of the keras feature-set (see https://tensorflow.rstudio.com/tools/installation.html#additional-dependencies)

```bash
pip install h5py pyyaml requests Pillow scipy
```

File: inst/examples/keras/cloudml.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+gcloud:
+  project         : "rstudio-cloudml"
+  account         : "javier@rstudio.com"
+  region          : "us-east1"
+  runtime-version : "1.2"
```
Comment (jjallaire) [inline inst/examples/keras/cloudml.yml]: See previous comment, I think this might belong in the `cloudml` key?

Comment (javierluraschi) [reply inst/examples/keras/cloudml.yml]: makes sense, made generic.

File: inst/cloudml/setup.py (hunk 1 above)
Comment (javierluraschi) [reply inst/cloudml/setup.py]: ah, great, added.

File: inst/examples/keras/cloudml.yml (hunk 2 above)
Comment (jjallaire) [reply inst/examples/keras/cloudml.yml]: I mean that the `runtime-version` key might belong in the `cloudml` rather the `gcloud` config section

**Comment by jjallaire** [comment]
If you do the pip install with the `--user` flag and then add the directory `$HOME/.cache/pip` to your cache then all of these installs will get cached (and scipy in particular can take a while). See my travis file for the Keras repo: https://github.com/rstudio/keras/blob/master/.travis.yml

**Comment by javierluraschi** [comment]
I'm going to merge this one and follow up on the pip and keras cache since it ended up being a good chunk of work.

**Pull request merged**